reference-implementation/lib/readable-stream.js

```diff
@@ -1156,12 +1156,22 @@ function SetUpReadableStreamDefaultControllerFromUnderlyingSource(stream, underl
     return InvokeOrNoop(underlyingSource, 'start', [stream._readableStreamController]);
   }
 
-  function pullAlgorithm() {
-    return PromiseInvokeOrNoop(underlyingSource, 'pull', [stream._readableStreamController]);
+  // eslint-disable-next-line func-style
```

Comment:
I see what you mean about the controller not existing yet being a problem for algorithm construction. Can we just move the single line

```js
const controller = Object.create(ReadableStreamDefaultController.prototype);
```

before the algorithm creation? Then perhaps pass that as a parameter to the SetUpXYZ() abstract ops? I guess maybe also move the line

```js
stream._readableStreamController = controller;
```

and then we can pass in `controller` instead of `stream` to the SetUpXYZ() ops.

Reply:
This worked, thanks. I chose to pass both stream and controller and still leave tying them together to SetUpXYZ().